sendsmaily.php

```diff
@@ -167,3 +167,88 @@ function smaily_subscribe_callback() {
 }
 add_action( 'wp_ajax_smaily_subscribe_callback', 'smaily_subscribe_callback' );
 add_action( 'wp_ajax_nopriv_smaily_subscribe_callback', 'smaily_subscribe_callback' );
+
+// Handles action for form submit in case of no js. Like free icegram plugin.
+function smaily_nojs_subscribe_callback() {
+	global $wpdb;
+
+	// Verify nonce.
+	if ( ! wp_verify_nonce( sanitize_key( $_POST['nonce'] ), 'smaily_nonce_field' ) ) {
+		wp_safe_redirect( home_url() );
+		return;
+	}
+
+	// Form data required.
+	if ( ! $_POST ) {
+		wp_safe_redirect( home_url() );
+		exit;
+	}
+
+	// Parse form data out of POST and sanitize.
+	$params = array();
+	foreach( $_POST as $key => $value ) {
+		if ( $key === "action" || $key === "nonce" ) {
+			continue;
+		}
+		$params[ $key ] = sanitize_text_field( $value );
+	}
+
+	// E-mail required.
+	if ( ! ( isset( $params['email'] ) && ! empty( $params['email'] ) ) ) {
+		wp_safe_redirect( home_url() );
+		exit;
+	}
+
+	// Get home url.
+	$home_url = home_url();
+
+	// Get data from database.
+	$table_name = esc_sql( $wpdb->prefix . 'sendsmaily_config' );
+	$config     = $wpdb->get_row( "SELECT * FROM `$table_name`" );
+
+	// Make a opt-in request to server.
+	$server = 'https://' . $config->domain . '.sendsmaily.net/api/opt-in/';
+	$lang   = explode( '-', $params['lang'] );
+	$array  = array(
+		'remote'        => 1,
+		'success_url'   => $home_url,
+		'failure_url'   => $home_url,
+		'language'      => $lang[0],
+	);
+
+	// Add autoresponder if selected.
+	if ( (int) $config->autoresponder !== 0 ) {
+		$array['autoresponder'] = $config->autoresponder;
+	}
+
+	$form_values = [];
+	// Add custom form values to Api request if available.
+	foreach ( $params as $key => $value ) {
+		if ( array_key_exists( $key, $array ) ) {
+			continue;
+		} else {
+			$form_values[ $key ] = $value;
+		}
+	}
+
+	$array = array_merge( $array, $form_values );
+	require_once( BP . DS . 'code' . DS . 'Request.php' );
+	$request = new Wp_Sendsmaily_Request( $server, $array );
+	$result  = $request->exec();
+
+	if ( empty( $result ) ) {
+		echo esc_html__( 'Something went wrong', 'wp_sendsmaily' );
+		wp_safe_redirect( home_url() );
```

Comment:
No need to have `wp_safe_redirect` inside the if (and elseif) condition when line 250 also calls `wp_safe_redirect`.

Reply:
fixed in 0821a75091dca26683fa0790c3c83128eda6d1ce